packages/pwa-kit-cli/src/ssr/server/build-dev-server.js

```diff
@@ -100,6 +99,9 @@ export const DevServerMixin = {
                 console.log(chalk.cyan('First build complete'))
             }, 75)
         })
+        if (config.some((cnf) => cnf.name === 'server')) {
+            app.__hotServerMiddleware = webpackHotServerMiddleware(app.__compiler)
+        }
```

Comment:
We haven't communicated the dependencies on these compiler names very well. But, I'm not sure I can see a fix that is better than this right now.

It'd be great if we could remain open to swapping the HMR implementation out in future! [Discussion here](https://mobify.slack.com/archives/C0DKK1FJS/p1651076799364259).